src/components/Upload/FileUploader.js

```diff
@@ -52,23 +86,63 @@ export default class FileUploader extends Component {
     return result;
   }
 
+  convertIPUZ(readerResult) {
+    const {grid, info, circles, shades, across, down} = iPUZtoJSON(readerResult);
```

Comment:
I tried to follow the pattern here that was done for converting to `puz`, so that way things would be fairly easy to follow.

Reply:
great!